crates/data-plane-controller/src/controller.rs

```diff
@@ -155,6 +164,8 @@ impl Controller {
             Status::AwaitDNS2 => self.on_await_dns_2(state).await?,
         };
 
+        validate_state(ops_checkout, state).await?;
```

Comment:
Not sure I understand the purpose of doing validation here. Haven't we already done the thing, this step? Shouldn't we just validate the next poll?

I suppose this is trying to catch invalid state transitions being attempted by `on_idle` pulling from `row_state`, but why not do that validation there, closer to where the transition is attempted?

Reply:
@jgraettinger we validate once before we operate on a state to make sure the data-plane-controller does not operate on a malformed state.

We then validate that after having done whatever step was necessary (pulumi up, ansible, etc.), the updated state as a result of these operations has not become malformed now, in effect preventing a malformed state from being written to `internal.tasks.inner_state`, to avoid having to manually fix malformed states